recipe/pyproject.toml.patch

```diff
@@ -0,0 +1,14 @@
+diff --git a/pyproject.toml b/pyproject.toml
```

Comment:
Is it any different if youuse 

git diff --patch 

To generate the patch

Reply:
`git diff` and `git diff --patch` give identical results.

Reply:
Use `git format-patch <hash_or_tag> --no-signature`

Reply:
Thanks, @h-vetinari .  I regenerated the patch: https://github.com/conda-forge/photutils-feedstock/pull/49/commits/3561a3d830e5e3c0d025f8e6cd50d055e833ab7f. 🤞

Reply:
Build is still failing.

Reply:
For reference, a similar patch file had no issues with a recent `astropy-feedstock` PR:
https://github.com/conda-forge/astropy-feedstock/pull/134/files#diff-4dc5d610c5dc06b54c2d280425d8b26ab43abbd64fe6d6cf3aa7b6ed8f3c9653

Is there some other issue with this PR?

Reply:
The logs tell you the issue:
```
| pyproject.toml | 2 +-
| 1 file changed, 1 insertion(+), 1 deletion(-)
|
|diff --git a/pyproject.toml b/pyproject.toml
|index 1200daa1..e56352e0 100644
|--- a/pyproject.toml
|+++ b/pyproject.toml
--------------------------
No file to patch.  Skipping patch.
```

You're trying to patch the source tarball from PyPI, which isn't necessarily equivalent to the git tag. The easiest way is just to change the source URL to the github tag directly.

Reply:
I understand that -- the `pyproject.toml` file *is* in the PyPI source tarball.  What I gleaned from the logs is that it appears the issue is that the patch level that is automatically determined is wrong.  It uses level 0, but it should be level 1.  For linux and win jobs it's using `-Np0` instead of `-Np1`.  For osx jobs, it uses `-Np1` and those pass.  Why does it select the wrong patch level, but only for linux/win?

```
Patch level ambiguous, selecting least deep
Applying patch: /home/conda/recipe_root/pyproject.toml.patch with args:
['-Np0', '-i', '/tmp/tmpdzdb757p/pyproject.toml.patch.native', '--binary']
can't find file to patch at input line 5
Perhaps you used the wrong -p or --strip option?
```

I'll try your suggestion, but I think the tarballs are identical.

Reply:
CI is still failing.  I verified that the PyPI and GH tarballs are identical.

The failure seems to be caused by using p0 instead of p1:

```
> wget https://pypi.io/packages/source/p/photutils/photutils-1.12.0.tar.gz
> tar xvfz photutils-1.12.0.tar.gz
> cd photutils-1.12.0
> wget https://raw.githubusercontent.com/conda-forge/photutils-feedstock/460fe27a1edfb1b76c2957ae830ac5d7115bed7e/recipe/pyproject.toml.patch
> patch --batch -Np0 -i pyproject.toml.patch
No file to patch.  Skipping...
1 out of 1 hunks ignored--saving rejects to a/pyproject.toml.rej
Can't create a/pyproject.toml.rej, output is in /var/folders/ly/xztygm217pd97f4m_4srfbw0000132/T/patchraK3c2QanNG: No such file or directory

> patch --batch -Np1 -i pyproject.toml.patch
patching file pyproject.toml
```

There does not appear to be a way to force patch level 1 to be used (https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#patches).

Reply:
@conda-forge/core I removed the patch to see if anything else was wrong.  CI is now failing (only on linux/win) with a different error:
```
ERROR: Directory '.' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.
```

`pyproject.toml` is definitely in the PyPI source tarball, so I've very confused.  Any ideas?

Reply:
```
> wget https://pypi.io/packages/source/p/photutils/photutils-1.12.0.tar.gz
> tar xvfz photutils-1.12.0.tar.gz
> ls photutils-1.12.0/pyproject.toml
photutils-1.12.0/pyproject.toml
```